### PR TITLE
Add pull request template

### LINKED ISSUE #361: Complete Community Standards checklist
To complete the Community Standards [checklist](https://github.com/istio-ecosystem/sail-operator/community), we need to add 2 items to our project:

- [X] Code of conduct - https://github.com/istio-ecosystem/sail-operator/pull/362
- [x] Pull request template - https://github.com/istio-ecosystem/sail-operator/pull/366

With this, we will meet the Opensource community standards

### PR DESCRIPTION
**Pull request by MaxBab**
The pull request template will help to add important information for the reviewers during new PR creation.

Fixes #361

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+ <!--  Thanks for sending a pull request!  Here are some tips for you:
+
+    1. If this is your first time, please read our contributor guidelines: https://github.com/istio-ecosystem/sail-operator/blob/main/CONTRIBUTING.md
+    2. Discuss your changes before you start working on them. You can open a new issue in the [Sail Operator GitHub repository](https://github.com/istio-ecosystem/sail-operator/issues) or start a discussion in the [Sail Operator Discussion](https://github.com/istio-ecosystem/sail-operator/discussions). By this way, you can get feedback from the community and ensure that your changes are aligned with the project goals.
+    3. If the PR is unfinished, make is as draft.
```
Comment (fjglira) [inline .github/pull_request_template.md:5]: Hey, maybe rephrase `If the PR is unfinished, mark it as a draft.`

**Review by fjglira — COMMENTED** [review]
lgtm, only a minor comment

**Comment by fjglira** [comment]
Hey @MaxBab can you check if we can set default labels to the PR based on the selection of the PR type? Or add just a note in the template to remember to add a label to identify the PR more easy

File: .github/pull_request_template.md (hunk 1 above)
Comment (MaxBab) [reply .github/pull_request_template.md]: Done

**Comment by MaxBab** [comment]
> Hey @MaxBab can you check if we can set default labels to the PR based on the selection of the PR type? Or add just a note in the template to remember to add a label to identify the PR more easy

I've checked the option to automatically label PRs.
All the solutions I found were a part of github actions.

Unfortunately, none of them is relevant for us.

There was one really cool action that is able to parse the PR subject and commit message context, but unfortunately, this action is abandoned for a really lond time, and has some dependencies which is already outdated.

Other active and working github action is able to set the label according to the directory structure.
So, where the change was made, the relevant label will be applied.

That is not so relevant for us, I believe.

All other similar actions abandoned.


At the end, I've added a note with relevant labels.

**Review by fjglira — APPROVED** [review]
lgtm
